File: u3u2/Assets/Scripts/app/net/c2s/CGSaveMail.cs
using System;
using System.IO;
namespace app.net
{

/**
 * 保存邮件
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGSaveMail :BaseMessage
{
	
	/** 要保存的邮件uuid */
	private string uuid;
	
	public CGSaveMail ()
	{
	}
	
	public CGSaveMail (
			string uuid )
	{
			this.uuid = uuid;
	}
	
	protected override void ReadImpl() 
	{
		return;
	}
	
	protected override void WriteImpl() 
	{
	// 要保存的邮件uuid
	WriteString(uuid);

	}
	
	public override short GetMessageType()
	{
		return (short)MessageType.CG_SAVE_MAIL;
	}
	
	public override string getEventType()
	{
		return "";
	}
	
	}
}

File: IdeaServer/zljnew/game_server/src/com/imop/lj/gameserver/mail/msg/CGSaveMail.java
package com.imop.lj.gameserver.mail.msg;

import com.imop.lj.core.msg.MessageType;
import com.imop.lj.gameserver.common.msg.CGMessage;
import com.imop.lj.gameserver.mail.handler.MailHandlerFactory;

/**
 * 保存邮件
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGSaveMail extends CGMessage{
	
	/** 要保存的邮件uuid */
	private String uuid;
	
	public CGSaveMail (){
	}
	
	public CGSaveMail (
			String uuid ){
			this.uuid = uuid;
	}
	
	@Override
	protected boolean readImpl() {

	// 要保存的邮件uuid
	String _uuid = readString();
	//end



			this.uuid = _uuid;
		return true;
	}
	
	@Override
	protected boolean writeImpl() {

	// 要保存的邮件uuid
	writeString(uuid);


		return true;
	}
	
	@Override
	public short getType() {
		return MessageType.CG_SAVE_MAIL;
	}
	
	@Override
	public String getTypeName() {
		return "CG_SAVE_MAIL";
	}

	public String getUuid(){
		return uuid;
	}
		
	public void setUuid(String uuid){
		this.uuid = uuid;
	}


	@Override
	public void execute() {
		MailHandlerFactory.getHandler().handleSaveMail(this.getSession().getPlayer(), this);
	}
}